st3/sublime_lib/show_selection_panel.py

```diff
@@ -0,0 +1,68 @@
+NOT_GIVEN = {}
+
+
+def show_selection_panel(
+    window,
+    items,
+    *,
+    flags=None,
+    labels=str,
+    selected=NOT_GIVEN,
+    on_select=None,
+    on_cancel=None,
+    on_highlight=None
+):
+    """Open a quick panel in the given window to select an item from a list.
+
+    :argument window: The :class:`sublime.Window` in which to show the panel.
+
+    :argument items: A :class:`list` of items to choose from.
+
+    Optional keyword arguments:
+
+    :argument flags: A bitwise OR of :const:`sublime.MONOSPACE_FONT` and
+    :const:`sublime.KEEP_OPEN_ON_FOCUS_LOST`.
+
+    :argument labels: A function taking elements of `items` to string labels.
+
+    :argument selected: The value in `items` that will be initially selected.
+    If `selected` is not given, no value will be initially selected.
+
+    :argument on_select: A callback accepting a value from `items` to be
+    invoked when the user chooses an item.
+
+    :argument on_cancel: A callback that will be invoked with no arguments if
+    the user closes the panel without choosing an item.
+
+    :argument on_highlight: A callback accepting a value from `items` to be
+    invoked every time the user changes the highlighted item in the panel.
+    """
+    labels_list = list(map(labels, items))
+
+    def on_done(index):
+        if index == -1:
+            if on_cancel:
+                on_cancel()
+        else:
```

Comment:
```suggestion
        elif on_select:
```

And remove the next line and unindent the following. No idea how to submit multi-line suggestions.